src/nb_func.cpp

```diff
@@ -575,14 +580,17 @@ static PyObject *nb_func_vectorcall_complex(PyObject *self,
 
             memset(kwarg_used, 0, nkwargs_in * sizeof(bool));
 
-            // 1. Copy positional arguments, potentially substitute kwargs/defaults
+            // 1. Copy individual arguments, potentially substitute kwargs/defaults
             size_t i = 0;
-            for (; i < nargs_pos; ++i) {
+            for (; i < nargs_step1; ++i) {
+                if (has_var_args && i == nargs_pos)
+                    continue; // skip nb::args argument, will be handled below
+
                 PyObject *arg = nullptr;
                 bool arg_convert  = pass == 1,
                      arg_none     = false;
 
-                if (i < nargs_in)
+                if (i < nargs_in && i < nargs_pos)
```

Comment:
didn't you already `continue` in the `i == nargs_po`s case just a few lines before?

Reply:
This is actually a quite important point, would you mind merging the comment into the code?

This was sort of the issue with the pybind11 version of the function call dispatcher -- it's a lot of tricky indexing. After a while, it became impossible understand the implementation without spending a significant amount of time on it -- comments like this go a long way in making the code more accessible.